### PR TITLE
(ISA-192) Region reports research effort graphs

### PR DESCRIPTION
**Pull request by JoshGx2000**
[ISA-192](https://jira.its.utas.edu.au/browse/ISA-192)

Adaptive research effort graphs for the region report data pages. Data for the graphs is queried and stored in the Django server media storage, where it is then retrieved via AJAX and displayed as a graph using Vega.

You cannot argue with the results:
Screen Shot 2022-10-19 at 3 38 35 pm

### CODE REVIEW

File: wordpress/plugins/region-report/single-region_report.php [hunk 1]
```diff
@@ -368,7 +370,137 @@ function addToList(list, text) {
...
+                    <div id="region-report-research-effort-<?php the_ID(); ?>-2"></div>
+                </div>
+                <script>
+                     postElement.addEventListener(
+                        "researchEffort",
+                        e => {
+                            const values = [];
+                            e.detail.forEach(e => {
+                                values.push({
+                                    year: e.year,
```
Comment (markhepburn) [inline wordpress/plugins/region-report/single-region_report.php:385]: You have two distinct bindings for "e" here, which might get confusing (sometimes it's a useful "trick" that makes code read clearer, but I worry here that it's hard to keep track of which e is what)

**Review by markhepburn — APPROVED** [review]
Minor comment, otherwise lgtm.  I know we briefly raised earlier whether it mightn't make as much sense to use the API endpoint, but I don't know that it's important (unless I'm forgetting something the only benefit is being able to control CORS headers ourselves, but that's a once-off fix anyway)

**Comment by JoshGx2000** [comment]
@markhepburn Will switch over to API endpoint in other cases, however for the case of the research effort graphs there's an issue with how long the queries take that makes caching the results in the media storage a more effective solution.

**Comment by markhepburn** [comment]
(just coming back to this, reviewing the last meeting notes) I think the main concern was ensuring that the data is the same between sidebar and report pages, which presumably means the API is the same where applicable.

**Pull request merged**